src/elements/code-snippet.svelte

```diff
@@ -0,0 +1,62 @@
+<svelte:options immutable tag='v-code-snippet' />
+
+<script lang='ts'>
+
+import pkg from '../../package.json';
+
+const version = pkg.devDependencies['prismjs'].replace('^', '');
+const cdn = (src: string) => `https://cdnjs.cloudflare.com/ajax/libs/prism/${version}/${src}`
+
+let language = ''
+
+const highlight = async (el: HTMLElement, language: string) => {
+  const { Prism } = window as { Prism: typeof import('prismjs') }
+
+  if (!Prism) {
+    await import(/* @vite-ignore */cdn('prism.min.js'))
+  }
+
+  await import(/* @vite-ignore */cdn(`components/prism-${language}.min.js`))
+
+  Prism.highlightElement(el)
+}
+
+export class VCode extends HTMLElement {
+  observer!: MutationObserver
```

Comment:
@micheal-parks I was able to get this to work as a Svelte file rather than Typescript, however I am getting a Svelte parse error for "observer." 

```
(property) VCode.observer: MutationObserver
Unexpected token

If you expect this syntax to work, here are some suggestions: 
If you use typescript with `svelte-preprocess`, did you add `lang="ts"` to your `script` tag? 
Did you setup a `svelte.config.js`? 
See https://github.com/sveltejs/language-tools/tree/master/docs#using-with-preprocessors for more info.svelte(parse-error)
```